ui/frontend/websocket.ts

```diff
@@ -0,0 +1,24 @@
+export default function openWebSocket() {
+  try {
+    const wsUri = ((window.location.protocol === "https:") ? "wss://" : "ws://") + window.location.host + "/websocket";
+    const socket = new WebSocket(wsUri);
+    return socket;
+  } catch (_) {
+    // WebSocket url error or WebSocket is not supported by browser.
+    // Assume it's the second case since url error is easy to notice.
+    (async () => {
+      try {
+        await fetch("/nowebsocket", {
+          method: 'post',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({})
```

Comment:
Do we need a body at all, since it's just empty?

Reply:
Not really. Will simplify this part.